### PR TITLE
Add a utility function for batching git commands

### PR DESCRIPTION
**Pull request by oleorhagen**
A lot of our code has 100+ lines of basically git boiler plate.

This adds a utility package, for executing batch git commands.

Just want to see if I should finish this, I think it'd be a nice addition?

### CODE REVIEW

**Review by lluiscampos — COMMENTED** [review]
Beautiful idea :heart: 

I'll review it properly once is ready, but this is clearly a thumbs up from me

File: cherrypicks.go [hunk 1]
```diff
@@ -49,52 +48,29 @@ func suggestCherryPicks(log *logrus.Entry, pr *github.PullRequestEvent, githubCl
...
-	countCmd.Dir = tmpdir
-	out, err = countCmd.CombinedOutput()
+	countCmd := exec.Command(
+		"sh", "-c",
+		"git log "+baseSHA+"...pr_"+prNumber+" | grep -i -e \"^    Changelog:\" | grep -v -i -e \"^    Changelog: *none\" | wc -l")
+	countCmd.Dir = state.Dir
+	out, err := countCmd.CombinedOutput()
 	if err != nil {
-		return fmt.Errorf("%v returned error: %s: %s", gitcmd.Args, out, err.Error())
+		return fmt.Errorf("%v returned error: %s: %s", countCmd.Args, out, err.Error())
```
Comment (lluiscampos) [inline cherrypicks.go:73]: Good catch

**Review by lluiscampos — APPROVED** [review]
Awesome work. Thank you.

It is the kind of PR where everything looks good but at the same time you know is easy to miss something... I hope we don't break any functionality :crossed_fingers:

**Comment by oleorhagen** [comment]
> Awesome work. Thank you.
> 
> It is the kind of PR where everything looks good but at the same time you know is easy to miss something... I hope we don't break any functionality crossed_fingers

Yes, agreed. It'll be interesting to see. Kept it tightly nit in one commit though, so reverting will be easy.

Giving it a spin now ;)

**Pull request merged**